src/coreclr/vm/excep.cpp

```diff
@@ -6533,59 +6533,48 @@ AdjustContextForJITHelpers(
 
 #if defined(USE_FEF) && !defined(TARGET_UNIX)
 
-struct SavedExceptionInfo
+static void FixContextForFaultingExceptionFrame(
+    EXCEPTION_POINTERS* ep,
+    EXCEPTION_RECORD* pOriginalExceptionRecord,
+    CONTEXT* pOriginalExceptionContext)
 {
-    EXCEPTION_RECORD m_ExceptionRecord;
-    CONTEXT m_ExceptionContext;
-    CrstStatic m_Crst;
-
-    void SaveExceptionRecord(EXCEPTION_RECORD *pExceptionRecord)
-    {
-        LIMITED_METHOD_CONTRACT;
-        size_t erSize = offsetof(EXCEPTION_RECORD, ExceptionInformation) +
-            pExceptionRecord->NumberParameters * sizeof(pExceptionRecord->ExceptionInformation[0]);
-        memcpy(&m_ExceptionRecord, pExceptionRecord, erSize);
-
-    }
+    WRAPPER_NO_CONTRACT;
 
-    void SaveContext(CONTEXT *pContext)
-    {
-        LIMITED_METHOD_CONTRACT;
-#ifdef CONTEXT_EXTENDED_REGISTERS
+    // don't copy param args as have already supplied them on the throw
+    memcpy((void*) ep->ExceptionRecord,
```

Comment:
What is `EXCEPTION_RECORD::ExceptionRecord` used for? Is always going to be NULL here or is it actually used for something?

Reply:
If it is always null, it may be worth it to add assert.

Reply:
It seems to be used for [chaining](https://docs.microsoft.com/en-us/cpp/dotnet/differences-in-exception-handling-behavior-under-clr) in case a native exception filter throws another exception:

> When an exception is raised during the processing of an [exception filter](https://docs.microsoft.com/en-us/cpp/cpp/writing-an-exception-filter?view=msvc-170) within managed code, the exception is caught and treated as if the filter returns 0.

> This is in contrast to the behavior in native code where a nested exception is raised, the ExceptionRecord field in the EXCEPTION_RECORD structure (as returned by [GetExceptionInformation](https://docs.microsoft.com/en-us/windows/win32/Debug/getexceptioninformation)) is set, and the ExceptionFlags field sets the 0x10 bit.

Again, I just restored the old code here.  I think it should always be `NULL` unless we have a bug in CLR.